experimental/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,355 @@
+# gives the common components of two affine plane curves
+
+@doc Markdown.doc"""
+    common_components(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem(C::AffinePlaneCurve)
+
+Return the affine plane curve consisting of the common component of `C` and `D`, or an empty vector if they do not have a common component.
+"""
+function common_components(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+  G = gcd(C.eq, D.eq)
+  if isone(G)
+     return Vector{AffinePlaneCurve}()
+  else
+     return [AffinePlaneCurve(G)]
+  end
+end
+
+################################################################################
+# Compute the intersection of two curves. The output is a list: the first
+# element the affine plane curve defined by the common components of the two
+# curves (or is empty if no common component), the second element is the list
+# of intersection points. Some might be contained in the common component too.
+
+@doc Markdown.doc"""
+    curve_intersect(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+
+Return a list whose first element is the affine plane curve defined by the gcd of `C.eq` and `D.eq`, the second element is the list of the remaining intersection points when the common components are removed from `C` and `D`.
+"""
+function curve_intersect(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+  G = gcd(C.eq, D.eq)
+  R = parent(C.eq)
+  if !isone(G)
+     # We divide by the gcd to get curves without common components.
+     F = div(C.eq, G)
+     H = div(D.eq, G)
+  else
+     F = C.eq
+     H = D.eq
+  end
+  I = ideal(R, [F, H])
+  # We eliminate the first variable in the ideal.
+  B = eliminate(I, [gen(R, 1)])
+  # We factorize the polynomial we obtain (which is a polynomial where only
+  # the second variable appears).
+  Z = factor(gen(B, 1))
+  Y = []
+  L = Array{S, 1}[]
+  # For the linear factors of g, we add the constant coefficient to the list Y.
+  # The constant coefficient is stored as a mpoly to be used later in evaluate.
+  for g in keys(Z.fac)
+     if total_degree(g) == 1
+        f = g//lc(g)
+        push!(Y, -f + gen(R, 2))
+     end
+  end
+  if !isempty(Y)
+     # For each y, we compute the possible values of x by replacing the second
+     # variable by the value y, and factorizing the resulting polynomial.
+     for y in Y
+        FF = evaluate(F, [gen(R, 1), y])
+        HH = evaluate(H, [gen(R, 1), y])
+        GG = gcd(FF, HH)
+        ZZ = factor(GG)
+        for g in keys(ZZ.fac)
+           if total_degree(g) == 1
+              f = g//lc(g)
+              # We use lc to convert the mpoly into element of S.
+              push!(L, [lc(-f + gen(R, 1)), lc(y)])
+           end
+        end
+     end
+  end
+  # L contains the intersection points as arrays (or can be empty).
+  Pts = Array{Point, 1}()
+  CC = Array{AffinePlaneCurve, 1}()
+  if !isempty(L)
+     for p in L
+        push!(Pts, Point(p))
+     end
+  end
```

Comment:
I *think* the following would work
```suggestion
  Pts = Point[Point(p) for p in L]
  CC = Vector{AffinePlaneCurve}()
```